runtime/include/chpl-mem-desc.h

```diff
@@ -39,110 +41,94 @@
-          "string move data"),                                          \
-        m(STRING_SELECT_DATA,                                           \
-          "string select data"),                                        \
-        m(CONFIG_ARG_COPY_DATA,                                         \
-          "config arg copy data"),                                      \
-        m(CONFIG_TABLE_DATA,                                            \
-          "config table data"),                                         \
-        m(LOCALE_NAME_BUFFER,                                           \
-          "locale name buffer"),                                        \
-        m(TASK_DESCRIPTOR,                                              \
-          "task descriptor"),                                           \
-        m(TASK_DESCRIPTOR_LINK,                                         \
-          "task descriptor link"),                                      \
-        m(TASK_STACK,                                                   \
-          "task stack"),                                                \
-        m(MUTEX,                                                        \
-          "mutex"),                                                     \
-        m(LOCK_REPORT_DATA,                                             \
-          "lock report data"),                                          \
-        m(TASK_POOL_DESCRIPTOR,                                         \
-          "task pool descriptor"),                                      \
-        m(TASK_LIST_DESCRIPTOR,                                         \
-          "task list descriptor"),                                      \
-        m(THREAD_PRIVATE_DATA,                                          \
-          "thread private data"),                                       \
-        m(THREAD_LIST_DESCRIPTOR,                                       \
-          "thread list descriptor"),                                    \
-        m(IO_BUFFER,                                                    \
-          "io buffer or bytes"),                                        \
-        m(OS_LAYER_TMP_DATA,                                            \
-          "OS layer temporary data"),                                   \
-        m(GMP,                                                          \
-          "gmp data"),                                                  \
-        m(GETS_PUTS_STRIDES,                                            \
-          "put_strd/get_strd array of strides"),                        \
-        m(GETS_PUTS_COUNTS,                                             \
-          "put_strd/get_strd array of count"),                          \
-        m(NUM, "")                      // this must be the last entry
+#define CHPL_MD_ALL_MEMDESCS(m)                                               \
```

Comment:
You could fit each definition on a single line w/o going over 80 characters. I think it makes it a little easier to read:

```c
#define CHPL_MD_ALL_MEMDESCS(m)                                               \
  m(UNKNOWN = 0,          "unknown",                               true),     \
  m(EXECUTION_COMMAND,    "chapel execution command buffer",       true),     \
  m(ARRAY_ELEMENTS,       "array elements",                        true),     \
  m(SET_WIDE_STRING,      "set wide string",                       true),     \
  m(GET_WIDE_STRING,      "get wide string",                       true),     \
  m(COMMAND_BUFFER,       "command buffer",                        true),     \
  m(COMM_XMIT_RCV_BUF,    "comm layer transmit/receive buffer",    true),     \
  m(COMM_FRK_SND_INFO,    "comm layer sent remote fork info",      true),     \
  m(COMM_FRK_SND_ARG,     "comm layer sent remote fork arg",       true),     \
  m(COMM_FRK_RCV_INFO,    "comm layer received remote fork info",  true),     \
  m(COMM_FRK_RCV_ARG,     "comm layer received remote fork arg",   true),     \
  m(COMM_FRK_DONE_FLAG,   "comm layer remote fork done flag(s)",   true),     \
  m(COMM_PER_LOC_INFO,    "comm layer per-locale information",     true),     \
  m(COMM_PRV_OBJ_ARRAY,   "comm layer private objects array",      true),     \
  m(COMM_PRV_BCAST_DATA,  "comm layer private broadcast data",     true),     \
  m(GLOM_STRINGS_DATA,    "glom strings data",                     true),     \
  m(STR_COPY_DATA,        "string copy data",                      true),     \
  m(STR_COPY_REMOTE,      "remote string copy",                    true),     \
  m(STR_CONCAT_DATA,      "string concat data",                    true),     \
  m(STR_MOVE_DATA,        "string move data",                      true),     \
  m(STR_SELECT_DATA,      "string select data",                    true),     \
  m(CFG_ARG_COPY_DATA,    "config arg copy data",                  true),     \
  m(CF_TABLE_DATA,        "config table data",                     true),     \
  m(LOCALE_NAME_BUF,      "locale name buffer",                    true),     \
  m(TASK_DESC,            "task descriptor",                       true),     \
  m(TASK_DESC_LINK,       "task descriptor link",                  true),     \
  m(TASK_STACK,           "task stack",                            true),     \
  m(MUTEX,                "mutex",                                 true),     \
  m(LOCK_REPORT_DATA,     "lock report data",                      true),     \
  m(TASK_POOL_DESC,       "task pool descriptor",                  true),     \
  m(TASK_LIST_DESC,       "task list descriptor",                  true),     \
  m(THREAD_PRV_DATA,      "thread private data",                   true),     \
  m(THREAD_LIST_DESC,     "thread list descriptor",                true),     \
  m(IO_BUFFER,            "io buffer or bytes",                    true),     \
  m(OS_LAYER_TMP_DATA,    "OS layer temporary data",               true),     \
  m(GMP,                  "gmp data",                              true),     \
  m(GETS_PUTS_STRIDES,    "put_strd/get_strd array of strides",    true),     \
  m(NUM,                  "this must be the last entry",           true)

```